dcicutils/deployment_utils.py

```diff
@@ -58,7 +64,282 @@ def boolean_setting(settings, key, default=None):
+        # to be high performance and it's convenient to organize logic like this.
+        configuration = list(filter(lambda d: (d.get('Value', '') != '' and d['OptionName'] not in ['AppSource'],
+                                    configuration)))
+
+        # upload the template
+        if is_cgap_env(env_name):
+            return eb_client.create_configuration_template(
+                ApplicationName=cls.EB_APPLICATION,
+                TemplateName=CGAP_ENV_INDEXER,
+                OptionSettings=configuration,
+                EnvironmentId=cls.extract_environment_id(env_name)
+            )
+        else:
+            return eb_client.create_configuration_template(
+                ApplicationName=cls.EB_APPLICATION,
+                TemplateName=FF_ENV_INDEXER,
+                OptionSettings=configuration,
+                EnvironmentId=cls.extract_environment_id(env_name)
+            )
+
+    @classmethod
+    def create_indexer_environment(cls, env_name, app_version):
+        """ Creates a new environment for indexing based on the given env_name. Will look for a template
+            called FF_ENV_INDEXER or CGAP_ENV_INDEXER, if that does not exist this call will fail.
+
+        :param env_name: env to base template off of
+        :param app_version: application version to deploy, MUST match that running on env_name
+        :return: result of EB env creation
+        :raises RuntimeError if bad env given
+        """
+        eb_client = boto3.client('elasticbeanstalk')
+        if is_cgap_env(env_name):
+            return eb_client.create_environment(
+                ApplicationName=cls.EB_APPLICATION,
+                EnvironmentName=CGAP_ENV_INDEXER,
+                TemplateName=CGAP_ENV_INDEXER,
+                VersionLabel=app_version
+            )[cls.STATUS] == cls.UPDATING and cls.delete_indexer_template(eb_client, FF_ENV_INDEXER)
+        elif is_fourfront_env(env_name):
+            return eb_client.create_environment(
+                ApplicationName=cls.EB_APPLICATION,
+                EnvironmentName=FF_ENV_INDEXER,
+                TemplateName=FF_ENV_INDEXER,
+                VersionLabel=app_version
+            )[cls.STATUS] == cls.UPDATING and cls.delete_indexer_template(eb_client, FF_ENV_INDEXER)
+        else:  # should never get here, but for good measure
+            raise RuntimeError('Tried to deploy indexer from an unknown environment: %s' % env_name)
+
+    @classmethod
+    def delete_indexer_template(cls, client, template_name):
+        """ Wrapper for "delete_configuration_template" that will only accept indexer_envs
+
+        :param client: boto3 elasticbeanstalk client
+        :param template_name: template to delete, will only accept indexer env templates
+        :return: True in success, False otherwise
+        """
+        if template_name not in [FF_ENV_INDEXER, CGAP_ENV_INDEXER]:
```

Comment:
It would be nice to have an `is_indexer_env` in `env_utils` rather than doing this at the naming level.